lib/eco/treemanager.py

```diff
@@ -1569,9 +1569,20 @@ def shrink_languagebox(self, lbox, newend):
         l = []
         n = newend
         while type(n) is not EOS:
-            l.append(n.symbol.name)
+            if isinstance(n, MultiTextNode):
+                for c in list(n.children):
+                    l.append(c.symbol.name)
+                    c.remove()
+                    c.deleted = True
+            else:
+                l.append(n.symbol.name)
             n.remove()
             n = n.next_term
+
+        for i in reversed(range(len(self.lines))):
```

Comment:
This looks dodgy: because we `del self.lines[i]` that means we'll skip some items and we could also get an `IndexError`. This should probably be something like:

```python
i = 0
while i < len(self.lines):
  if self.lines[i].node.deleted:
    del self.lines[i]
  else:
    i += 1
```

Reply:
I don't think we can skip items or get index errors since I'm running backwards through the list (note the `reversed`). But maybe your variant is more readable?

Reply:
Aha, you're right -- I missed the `reversed`. My version is probably more readable (although, ironically, it is slightly slower), in the sense that it's the more common idiom.

Reply:
So should I change it or leave it?

Reply:
Up to you.

Reply:
Leave it is then. ;-)